repose-aggregator/components/filters/herp-filter/src/test/scala/org/openrepose/filters/herp/HerpFilterTest.scala

```diff
@@ -387,6 +428,20 @@ class HerpFilterTest extends FunSpec with BeforeAndAfterEach with Matchers {
       logEvents.size shouldBe 1
       logEvents.get(0).getMessage.getFormattedMessage should include("\"ProjectID\" : [  \"foo\"  ,\"bar\"  ]")
     }
+    it("should extract and log multiple project id header values from Response if not available from Request") {
+      // given:
+      servletResponse.addHeader("X-Project-Id", "foo")
+      servletResponse.addHeader("X-Project-Id", "bar")
+
+      // when:
+      herpFilter.configurationUpdated(herpConfig)
+      herpFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+      // then:
+      def logEvents = listAppenderPre.getEvents
+      logEvents.size shouldBe 1
+      logEvents.get(0).getMessage.getFormattedMessage should include("\"ProjectID\" : [  \"foo\"  ,\"bar\"  ]")
+    }
```

Comment:
I'd love to see a test around the behavior if either the `X-Tenant-Id` or `X-Project-Id` header is present in both the request and response.